### PR TITLE
Use GnuPG to ensure file integrity

### PR DESCRIPTION
**Pull request by mconintet**
Hi,

Here are changes on this pull request:
1. Update submodule openssl to it's lastest master
2. Update libssh2 version to 1.7.0
3. Use HTTPS to download files
4. Use GnuPG to ensure file integrity and the public key is from here: https://www.libssh2.org/mail/libssh2-devel-archive-2014-04/0000.shtml

### CODE REVIEW

**Comment by x2on** [comment]
Did you check if 1.7.0 works fine on iOS?

**Comment by mconintet** [comment]
Hi,

No more tests yet but I built them with your example project it looks good :)

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,7 @@
-[submodule "openssl"]
-	path = openssl
-	url = https://github.com/x2on/OpenSSL-for-iPhone.git
 [submodule "libgcrypt-for-ios"]
 	path = libgcrypt-for-ios
 	url = https://github.com/x2on/libgcrypt-for-ios.git
+[submodule "openssl"]
+	path = openssl
+	url = https://github.com/mconintet/OpenSSL-for-iPhone.git
+	branch = patch-1
```
Comment (x2on) [inline .gitmodules:7]: please revert

**Comment by x2on** [comment]
the build is red

**Pull request closed without merging**